File: Assets/Game/Scripts/Services/Library/Encrypter.cs
using System;
using System.IO;
using System.Linq;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
using UnityEngine.Networking;

// 暗号化装置
// 通信路の入出力をチェックし、暗号化と復号化を行います。
public class Encrypter {
    //-------------------------------------------------------------------------- 初期化
    // TODO
    //string securitySetting = null;

    //-------------------------------------------------------------------------- 初期化
    // コンストラクタ
    public Encrypter(string securitySetting) {
        Init(securitySetting);
    }

    //-------------------------------------------------------------------------- 実装ポイント (初期化とクリア)
    // 初期化
    protected virtual void Init(string securitySetting) {
        // TODO
        //this.securitySetting = securitySetting;
        Clear();
    }

    // クリア
    protected virtual void Clear() {
        // TODO
        // this.securitySetting の値に応じて
        // エンクリプターを初期化
    }

    //-------------------------------------------------------------------------- 暗号化と復号化
    // 暗号化
    public string Encrypt(string message) {
        // TODO
        // this.securitySetting の値に応じて
        // 暗号化ロジックの実装
        return message;
    }

    // 復号化
    public string Decrypt(string message) {
        // TODO
        // this.securitySetting の値に応じて
        // 復号化ロジックの実装
        return message;
    }
}

File: Services/services/library/encrypter.js
var util = require('util');

// constructor
function Encrypter(config, logger) {
    this.init(config, logger);
}
util.inherits(Encrypter, function(){});

// init
Encrypter.prototype.init = function(config, logger) {
    this.config = config;
    this.logger = logger;

    // clear
    this.clear();
};

// clear
Encrypter.prototype.clear = function() {
    // NOTE
    // implement by inherit
}

// encrypt (outgoing)
Encrypter.prototype.encrypt = function(message) {
    return message;
}

// decrypt (incoming)
Encrypter.prototype.decrypt = function(message) {
    return message;
}

// exports
module.exports = Encrypter;
